### PR TITLE
Issue #7475: resolve Cognitive Complexity cases above 20

### PR DESCRIPTION
**Pull request by romani**
Issue #7475 

we need to verify that Sonar does reports such cases as violation.
but at https://sonarcloud.io/project/issues?id=org.checkstyle%3Acheckstyle&pullRequest=7478 I see not issues at all, looks like Sonar filter out all issues that present in master, so I do not know how to make sure I actually did fix for violation.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/Main.java [hunk 1]
```diff
@@ -786,24 +786,32 @@ else if (hasConfigurationFile) {
                     final String msg = "Could not find config XML file '%s'.";
                     result.add(String.format(Locale.ROOT, msg, configurationFile));
                 }
-
-                // validate optional parameters
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: why start here and not include the try/catch above? Then the whole new method could be called like `validateCliWithConfigurationFile` or such.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: config file is not optional, all other parameters are optional.
I think it is reasonable in current form, please review again.

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: I was referring to this is all part of the if.
> else if (hasConfigurationFile) {

Moving the try/catch to the new method would make the code:
````
else if (hasConfigurationFile) {
    result.addAll(validateOptionalCliParameters());
}
````

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: Config is not optional in comparison to MT parameters

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: sounds fine to me. I was only speaking about putting more code into the method.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: renamed, with more code ... it become hard to make name, .... we will think about it later.

**Pull request merged**